core/src/cluster_info.rs

```diff
@@ -450,23 +450,24 @@ impl ClusterInfo {
     /// since. This allows the bank to query for new votes only.
     ///
     /// * return - The votes, and the max timestamp from the new set.
-    pub fn get_votes(&self, since: u64) -> (Vec<Transaction>, u64) {
+    pub fn get_votes(&self, since: u64) -> (Vec<CrdsValueLabel>, Vec<Transaction>, u64) {
         let votes: Vec<_> = self
             .gossip
             .crds
             .table
-            .values()
-            .filter(|x| x.insert_timestamp > since)
-            .filter_map(|x| {
+            .iter()
+            .filter(|(_, x)| x.insert_timestamp > since)
+            .filter_map(|(label, x)| {
                 x.value
                     .vote()
-                    .map(|v| (x.insert_timestamp, v.transaction.clone()))
+                    .map(|v| (label.clone(), x.insert_timestamp, v.transaction.clone()))
             })
             .collect();
-        let max_ts = votes.iter().map(|x| x.0).max().unwrap_or(since);
-        let txs: Vec<Transaction> = votes.into_iter().map(|x| x.1).collect();
+        let max_ts = votes.iter().map(|x| x.1).max().unwrap_or(since);
+        let (labels, txs): (Vec<CrdsValueLabel>, Vec<Transaction>) =
+            votes.into_iter().map(|x| (x.0, x.2)).unzip();
```

Comment:
Can we just unzip in the first collect, so we don't have to shuffle around the transactions and labels again?

Reply:
Also this max calculation will be faster with a vec of packed timestamps.

Reply:
ah yea, of course, thanks!